### PR TITLE
ci: onboarding module to ibm registery

### PR DESCRIPTION
**Pull request by rajatagarwal-ibm**
### Description

Onboard Redis module to the IBM Registry

### Release required?

- [ ] No release
- [x] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

File: ibm_catalog.json [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+      "long_description": "Use this module to provision and configure an IBM Cloud Database for Redis instance.",
+      "offering_docs_url": "https://github.com/terraform-ibm-modules/terraform-ibm-icd-redis/blob/main/README.md",
+      "offering_icon_url": "https://raw.githubusercontent.com/terraform-ibm-modules/terraform-ibm-icd-redis/main/images/redis_icon.svg",
+      "features": [
+        {
+          "title": "Create Redis instance",
+          "description": "Create IBM Cloud Database for Redis instance."
+        },
+        {
+          "title": "Attach Users",
```
Comment (ocofaigh) [inline ibm_catalog.json:28]: Managing Users and Roles

File: ibm_catalog.json [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+          "description": "Attach list of users to the Redis instance."
```
Comment (ocofaigh) [inline ibm_catalog.json:29]: Manage IAM service credentials as well as redis admin and default user credentials and roles

File: ibm_catalog.json [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+          "title": "Create Service Credentials Roles",
+          "description": "Create Service Credentials Roles for the ICD Redis database instance."
+        },
+        {
+          "title": "KMS encryption",
+          "description": "Supports key management encryption (BYOK and KYOK), including service-to-service authorization policies with KMS."
+        },
+        {
+          "title": "CBR",
+          "description": "Create context-based restriction (CBR) rules for both instance and buckets."
```
Comment (ocofaigh) [inline ibm_catalog.json:45]: copy and paste issue - CBR rules are for redis instance

File: ibm_catalog.json [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+        }
```
Comment (ocofaigh) [inline ibm_catalog.json:46]: Suggest adding these features too:
- Redis configuration settings to tune your databases to your use-case
- Scaling Disk, Memory, and CPU (adjust the amount of resources available to your IBM Cloud® Databases for Redis deployment to suit your workload and the size of your data) **_[NOTE: This is manual scaling, not auto scaling which module also supports and you have included already]_**

**Review by ocofaigh — CHANGES_REQUESTED** [review]
see my comments

File: ibm_catalog.json [hunk 5]
```diff
@@ -0,0 +1,103 @@
+{
+  "products": [
+    {
+      "name": "terraform-ibm-icd-redis",
+      "label": "ICD Redis Module",
```
Comment (SirSpidey) [inline ibm_catalog.json:5]:
```suggestion
      "label": "ICD Redis module",
```

File: ibm_catalog.json [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+        "dev_ops",
+        "target_terraform",
+        "terraform",
+        "module"
+      ],
+      "keywords": [
+        "terraform",
+        "redis",
+        "icd"
+      ],
```
Comment (SirSpidey) [inline ibm_catalog.json:17]: add "database" for searching?

File: ibm_catalog.json [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+      "short_description": "Implements an instance of IBM Cloud Databases for Redis.",
+      "long_description": "Use this module to provision and configure an IBM Cloud Database for Redis instance.",
```
Comment (SirSpidey) [inline ibm_catalog.json:19]:
```suggestion
      "long_description": "Use this module to provision and configure an instance of IBM Cloud Databases for Redis.",
```

File: ibm_catalog.json [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+      "offering_docs_url": "https://github.com/terraform-ibm-modules/terraform-ibm-icd-redis/blob/main/README.md",
+      "offering_icon_url": "https://raw.githubusercontent.com/terraform-ibm-modules/terraform-ibm-icd-redis/main/images/redis_icon.svg",
+      "features": [
+        {
+          "title": "Create Redis instance",
+          "description": "Create IBM Cloud Database for Redis instance."
```
Comment (SirSpidey) [inline ibm_catalog.json:25]:
```suggestion
          "title": "Create instance of Databases for Redis",
          "description": "Creates an instance of IBM Cloud Databases for Redis."
```

File: ibm_catalog.json (hunk 1 above)
Comment (SirSpidey) [reply ibm_catalog.json:28]: just don't capitalize Users or Roles unless at beginning of sentence

File: ibm_catalog.json (hunk 2 above)
Comment (SirSpidey) [reply ibm_catalog.json:29]: Always refer to Redis either as `Databases for Redis` or `IBM Cloud Databases for Redis`

File: ibm_catalog.json [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+          "description": "Rules to allow the Redis database instance to increase resources in response to usage."
+        },
+        {
+          "title": "Create Service Credentials Roles",
+          "description": "Create Service Credentials Roles for the ICD Redis database instance."
+        },
+        {
+          "title": "KMS encryption",
+          "description": "Supports key management encryption (BYOK and KYOK), including service-to-service authorization policies with KMS."
+        },
```
Comment (SirSpidey) [inline ibm_catalog.json:42]:
```suggestion
          "title": "Auto scaling instance",
          "description": "Supports rules that allow the Cloud Databases instance to increase resources in response to usage."
        },
        {
          "title": "Create service credentials roles",
          "description": "Create service credentials roles for the instance of Databases for Redis."
        },
        {
          "title": "KMS encryption",
          "description": "Supports key management encryption (BYOK and KYOK), including service-to-service authorization policies with KMS."
        },
```

What are service credential roles? Are they really service credentials or can you create roles with different permissions?

File: ibm_catalog.json (hunk 4 above)
Comment (SirSpidey) [reply ibm_catalog.json:46]: Consider putting auto-scaling and manual scaling together in one feature if it makes sense (e.g., `Supports both manual and auto-scaling to configure the Cloud Databases deployment to your needs.`)

File: ibm_catalog.json [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+                  "url": "https://upload.wikimedia.org/wikipedia/commons/1/1d/No_image.svg",
+                  "type": "image/svg+xml"
+                },
+                "description": " "
+              }
+            ]
+          }
+        },
+        {
+          "label": "Complete example with BYOK encryption, and CBR rules",
```
Comment (SirSpidey) [inline ibm_catalog.json:67]:
```suggestion
          "label": "BYOK encryption and CBR rules",
```
Might provide enough context and be a shorter label.

**Review by SirSpidey — CHANGES_REQUESTED** [review]
Use either `Databases for Redis`, `IBM Cloud Databases for Redis` or `Cloud Databases` when you refer to redis.

File: ibm_catalog.json (hunk 6 above)
Comment (ocofaigh) [reply ibm_catalog.json:17]: perhaps we also add other redis capabilities such as `cache` and `message-broker` ?

File: ibm_catalog.json (hunk 9 above)
Comment (ocofaigh) [reply ibm_catalog.json:42]: A service credential can be assigned an IAM role such as `Administrator`, `Operator`, `Viewer` or `Editor`

File: ibm_catalog.json (hunk 10 above)
Comment (ocofaigh) [reply ibm_catalog.json:67]: `BYOK KMS encryption and CBR rules` maybe

**Comment by rajatagarwal-ibm** [comment]
Addressed all comments

**Comment by rajatagarwal-ibm** [comment]
> Consider putting auto-scaling and manual scaling together in one feature if it makes sense (e.g., Supports both manual and auto-scaling to configure the Cloud Databases deployment to your needs.)

@SirSpidey lets keep them as different features as the individual description of them is much helpful.

File: ibm_catalog.json (hunk 6 above)
Comment (ocofaigh) [reply ibm_catalog.json]: @rajatagarwal-ibm You may have missed my comment?

Comment (rajatagarwal-ibm) [reply ibm_catalog.json]: my bad, perhaps didn't refresh my browser. Addressed

File: ibm_catalog.json [hunk 11]
```diff
@@ -0,0 +1,114 @@
+{
+  "products": [
+    {
+      "name": "terraform-ibm-icd-redis",
+      "label": "ICD Redis module",
```
Comment (SirSpidey) [inline ibm_catalog.json:5]: I missed this. Need to refer to this as `IBM Cloud Databases for Redis module` or `Databases for Redis module`

File: ibm_catalog.json [hunk 12]
```diff
@@ -0,0 +1,114 @@
...
+          "title": "Create service credentials roles",
+          "description": "Create service credentials roles for the instance of Databases for Redis."
+        },
+        {
+          "title": "Configure Databases for Redis instance",
+          "description": "Configure Databases for Redis settings to tune database based on the use-case."
+        },
+        {
+          "title": "Scaling Disk, Memory, and CPU",
+          "description": "Adjust the amount of resources available to Database for Redis deployment to suit workload and the size of data."
```
Comment (SirSpidey) [inline ibm_catalog.json:48]:
```suggestion
          "title": "Scaling disk, memory, and CPU",
          "description": "Adjust the Cloud Databases resources to suit workloads and other needs."
```

**Review by SirSpidey — DISMISSED** [review]
missed the overall title. suggestions for additional features

File: README.md [hunk 13]
```diff
@@ -22,7 +22,7 @@ provider "ibm" {
 
 module "redis" {
   # replace "main" with a GIT release version to lock into a specific release
-  source            = "terraform-ibm-modules/icd-redis/ibm"
+  source            = "terraform-ibm-modules/terraform-ibm-icd-redis/ibm?ref=main"
```
Comment (ocofaigh) [inline README.md:25]: Please change the source back, it was already correct. This is wrong

Comment (ocofaigh) [reply README.md:25]: its missing the version: `version = "X.X.X" # Replace "X.X.X" with a release version to lock into a specific release`

File: README.md [hunk 14]
```diff
@@ -22,7 +22,7 @@ provider "ibm" {
 
 module "redis" {
   # replace "main" with a GIT release version to lock into a specific release
```
Comment (ocofaigh) [inline README.md:24]: remove this

File: ibm_catalog.json [hunk 15]
```diff
@@ -0,0 +1,114 @@
...
+      "keywords": [
+        "terraform",
+        "redis",
+        "icd",
+        "database",
+        "cache",
+        "message-broker"
+      ],
+      "short_description": "Implements an instance of IBM Cloud Databases for Redis.",
+      "long_description": "Use this module to provision and configure an instance of IBM Cloud Databases for Redis.\n\n#### Usage\n```hcl\nprovider \"ibm\" {\n  ibmcloud_api_key = \"XXXXXXXXXX\"\n  region           = \"us-south\"\n}\n\nmodule \"redis\" {\n  # Replace “X.X.X” with a release version to lock into a specific release\n  source            = \"https://cm.globalcatalog.cloud.ibm.com/api/v1-beta/offering/source?archive=tgz&kind=terraform&name=terraform-ibm-icd-redis&version=X.X.X\"\n  resource_group_id = \"xxXXxxXXxXxXXXXxxXxxxXXXXxXXXXX\"\n  region            = \"us-south\"\n  instance_name     = \"my-redis-instance\"\n}\n```",
```
Comment (ocofaigh) [inline ibm_catalog.json:22]: Usage should be a H3 heading

File: ibm_catalog.json [hunk 16]
```diff
@@ -0,0 +1,114 @@
...
+      "flavors": [
+        {
+          "label": "Basic",
+          "name": "basic",
+          "working_directory": "examples/basic",
+          "architecture": {
+            "diagrams": [
+              {
+                "diagram": {
+                  "caption": " ",
```
Comment (ocofaigh) [inline ibm_catalog.json:68]: Basic Example

File: ibm_catalog.json [hunk 17]
```diff
@@ -0,0 +1,114 @@
...
+                  "url": "https://upload.wikimedia.org/wikipedia/commons/1/1d/No_image.svg",
```
Comment (ocofaigh) [inline ibm_catalog.json:69]: https://raw.githubusercontent.com/terraform-ibm-modules/documentation/main/icons/ModuleExampleDiagram.svg

File: ibm_catalog.json [hunk 18]
```diff
@@ -0,0 +1,114 @@
...
+                  "type": "image/svg+xml"
+                },
+                "description": " "
```
Comment (ocofaigh) [inline ibm_catalog.json:72]: The description here should match the description at the top of the example readme

File: ibm_catalog.json [hunk 19]
```diff
@@ -0,0 +1,114 @@
...
+          "working_directory": "examples/complete",
+          "architecture": {
+            "diagrams": [
+              {
+                "diagram": {
+                  "caption": " ",
+                  "url": "https://upload.wikimedia.org/wikipedia/commons/1/1d/No_image.svg",
+                  "type": "image/svg+xml"
+                },
+                "description": " "
```
Comment (ocofaigh) [inline ibm_catalog.json:89]: same as above, these all need to be filled in

File: ibm_catalog.json [hunk 20]
```diff
@@ -0,0 +1,114 @@
...
+          "working_directory": "examples/fscloud",
+          "architecture": {
+            "diagrams": [
+              {
+                "diagram": {
+                  "caption": " ",
+                  "url": "https://upload.wikimedia.org/wikipedia/commons/1/1d/No_image.svg",
+                  "type": "image/svg+xml"
+                },
+                "description": " "
```
Comment (ocofaigh) [inline ibm_catalog.json:106]: same as above, these all need to be filled in

**Review by ocofaigh — CHANGES_REQUESTED** [review]
I left some comments. We also need to do some examples cleanup..

- Rename complete -> advanced
- Update all of the example readmes so they all have a one liner descrption at the top, followed by a bullet point list of things it deploys. For example, see https://github.com/terraform-ibm-modules/terraform-ibm-cos/tree/main/examples/advanced

**Comment by rajatagarwal-ibm** [comment]
Skipping upgrade tests as there is change in name of the example:

```
Error Trace:	/go/pkg/mod/github.com/gruntwork-io/terratest@v0.46.6/modules/terraform/destroy.go:11
        	            				/go/pkg/mod/github.com/terraform-ibm-modules/ibmcloud-terratest-wrapper@v1.23.16/testhelper/tests.go:345
        	            				/go/pkg/mod/github.com/terraform-ibm-modules/ibmcloud-terratest-wrapper@v1.23.16/testhelper/tests.go:577
        	            				/__w/terraform-ibm-icd-redis/terraform-ibm-icd-redis/tests/pr_test.go:101
        	Error:      	Received unexpected error:
        	            	FatalError{Underlying: error while running command: chdir /tmp/terraform-base-redis-advanced-upg-v2mxvv1105543843/examples/advanced: no such file or directory; }
        	Test:       	TestRunAdvancedExampleUpgrade
```

File: examples/advanced/README.md [hunk 21]
```diff
@@ -0,0 +1,12 @@
+# Advanced example
+
+This example creates a IBM Cloud Database for Redis instance with KMS encryption enabled and CBR rules configured.
+
+The following resources are provisioned by this example:
+
+- A new resource group, if an existing one is not passed in.
+- A basic VPC and subnet.
+- A Key Protect instance with a root key in the given resource group and region.
+- An instance of Databases for Redis with encryption enabled.
```
Comment (ocofaigh) [inline examples/advanced/README.md:10]: "with encryption enabled" -> "with KMS encryption enabled"

File: examples/fscloud/README.md [hunk 22]
```diff
@@ -1,19 +1,21 @@
-# Financial Services Cloud profile example
+# Financial Services compliant example
 
-An end-to-end example that uses the [Profile for IBM Cloud Framework for Financial Services](../../modules/fscloud/) to deploy an instance of IBM Cloud Databases for Redis.
+This example uses the [Profile for IBM Cloud Framework for Financial Services](../../modules/fscloud/) to provision an IBM Cloud Databases for Redis instance.
```
Comment (ocofaigh) [inline examples/fscloud/README.md:3]: you will need to update the hyperlink to full url (aka `https://github.com/terraform-ibm-modules/terraform-ibm-icd-redis/tree/main/modules/fscloud`) for it to link in registry

**Review by ocofaigh — CHANGES_REQUESTED** [review]
see comments

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**